test/integration/extension_discovery_integration_test.cc

```diff
@@ -215,13 +259,11 @@ class ExtensionDiscoveryIntegrationTest : public Grpc::GrpcClientIntegrationPara
     // Response with initial LDS.
     sendLdsResponse("initial");
 
-    // Wait for ECDS stream.
-    auto& ecds_upstream = getEcdsFakeUpstream();
-    result = ecds_upstream.waitForHttpConnection(*dispatcher_, ecds_connection_);
-    RELEASE_ASSERT(result, result.message());
-    result = ecds_connection_->waitForNewStream(*dispatcher_, ecds_stream_);
-    RELEASE_ASSERT(result, result.message());
-    ecds_stream_->startGrpcStream();
+    waitForEcdsStream(getEcdsFakeUpstream(), ecds_connection_, ecds_stream_);
```

Comment:
Previously, for LEDS, I remember I had a race between the 2 incoming ECDS streams. When registering for 2 resources, sometimes resource-A was received on stream-1, and sometimes resource-B was received on stream-1. Did you encounter something similar on the test that uses 2 streams?

One way to check (though not guaranteed) is to run the test 1000 times and see if it is flaky.

Reply:
In this case, the the two ECDS subscription are on two completely different connections. Each subscription has it's own cluster and upstream. I guess that's the reason it doesn't race.  BTW the logic is also in https://github.com/envoyproxy/envoy/blob/dede92e6e5888cdf27320bf3b6bc9987657d6e22/test/integration/listener_extension_discovery_integration_test.cc#L130, which had been in code base for a while.

I run the 1000 tests anyway, and doesn't flaky.